rdm/0006-profiles-and-flavours.md

```diff
@@ -0,0 +1,104 @@
+- Start Date: 2019-10-02
+- RFC PR: [#6](https://github.com/inveniosoftware/rfcs/pull/6)
+- Authors: Guillaume Viger
+
+# Customize created Invenio instance by choosing profile and optionally flavour
+
+## Summary
+
+When running the instance bootstrapping CLI command (`invenio-cli init`), the
+hoster should be given a choice to create an Integrated Library System (ILS) or
+Resource Data Management (RDM) "flavour" Invenio application. If RDM is chosen,
+there are further "profiles" to be chosen from, for instance "default" and
+"biomed". These choices pre-configure the application to varying levels.
+
+## Motivation
+
+> Why are we doing this? What user stories does it support? What is the expected
+outcome?
+
+As a (prospecting-)hoster, I want to have a quick working application for my
+main use case: library system or data management, so I can evaluate it
+right away (prospecting) and/or start using it (configuration/development). The
+RDM choice implies different kinds of data management for which I want to leverage
+pre-configuration / pre-customizations in order to get started faster and adopt
+same settings as other installations in my domain.
+
+As an InvenioRDM (core-, extension-, plugin-)developer for domain stakeholder,
+I want to be able to satisfy the exclusive needs/wants of my specialized
+stakeholders while still being able to develop and use domain authorities **within**
+the InvenioRDM context. I have a dedicated mechanism and space to insert
+customizations shared by instances for my domain this way.
+
+
+## Terminology
```

Comment:
The terminology is used before. Shall it be defined earlier?

Reply:
The summary will inevitably use the terms, but I think they are understandable from context there. 

I like presenting definitions only after the motivation has been laid out: the motivation is presented in the problem-domain (plain english) and the definitions are laid out in the solution-domain (technical specifications). In other words, first we motivate the issue in a relatable fashion, then we use specific terms in detailed design to solve it.

Reply:
Hmm, I haven't thought of it. Currently the cli is saving some config in a `.invenio` hidden file. However, which scenario do you foresee where filling all fields in the config instead of interactive filling them? Some sort of "automated" deployment (e.g. puppet, ansible)?

I would keep it interactive for now. We can get feedback and if people need it implement it. It wouldn't change much current code, mostly add.